src/EditorFeatures/CSharp/Formatting/CSharpEditorFormattingService.cs

```diff
@@ -125,53 +125,11 @@ private IEnumerable<AbstractFormattingRule> GetFormattingRules(Document document
         {
             var workspace = document.Project.Solution.Workspace;
             var formattingRuleFactory = workspace.Services.GetService<IHostDependentFormattingRuleFactoryService>();
-            return formattingRuleFactory.CreateRule(document, position).Concat(GetTypingRules(document, tokenBeforeCaret)).Concat(Formatter.GetDefaultFormattingRules(document));
+            return formattingRuleFactory.CreateRule(document, position).Concat(GetTypingRules(tokenBeforeCaret)).Concat(Formatter.GetDefaultFormattingRules(document));
         }
 
-        public async Task<IList<TextChange>> GetFormattingChangesOnReturnAsync(Document document, int caretPosition, CancellationToken cancellationToken)
-        {
-            var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
-            if (!options.GetOption(FeatureOnOffOptions.AutoFormattingOnReturn))
-            {
-                return null;
-            }
-
-            // first, find the token user just typed.
-            SyntaxToken token = await GetTokenBeforeTheCaretAsync(document, caretPosition, cancellationToken).ConfigureAwait(false);
-            if (token.IsMissing)
-            {
-                return null;
-            }
-
-            var formattingRules = this.GetFormattingRules(document, caretPosition, token);
-
-            string text = null;
-            if (IsInvalidToken(token, ref text))
-            {
-                return null;
-            }
-
-            // Check to see if the token is ')' and also the parent is a using statement. If not, bail
-            if (TokenShouldNotFormatOnReturn(token))
-            {
-                return null;
-            }
-
-            // if formatting range fails, do format token one at least
-            var changes = await FormatRangeAsync(document, token, formattingRules, cancellationToken).ConfigureAwait(false);
-            if (changes.Count > 0)
-            {
-                return changes;
-            }
-
-            // if we can't, do normal smart indentation
-            return await FormatTokenAsync(document, token, formattingRules, cancellationToken).ConfigureAwait(false);
-        }
-
-        private static bool TokenShouldNotFormatOnReturn(SyntaxToken token)
-        {
-            return !token.IsKind(SyntaxKind.CloseParenToken) || !token.Parent.IsKind(SyntaxKind.UsingStatement);
```

Comment:
this truly was an entire IEditorFormattingService that was created solely for hitting enter after the `)` in `using (a)`.  Very very strange!